src/main/java/org/checkerframework/checker/dependencyinjection/DependencyInjectionTransfer.java

```diff
@@ -48,31 +47,68 @@ public TransferResult<CFValue, CFStore> visitMethodInvocation(
               boundClassTypeMirror.getAnnotation(), diATF.classValValueElement, String.class);
 
       accumulate(node, result, classNames.toArray(new String[1]));
+
     } else if (diATF.isAnnotatedWithMethod(node.getTree())) {
+
       MethodInvocationNode methodInvocationNode = node;
-      MethodInvocationNode callToBindNode =
-          (MethodInvocationNode) methodInvocationNode.getTarget().getReceiver();
-      MethodInvocationNode namesClassNode =
-          (MethodInvocationNode) methodInvocationNode.getArgument(0);
-      FieldAccessNode boundClass = (FieldAccessNode) callToBindNode.getArgument(0);
-      StringLiteralNode givenName = (StringLiteralNode) namesClassNode.getArgument(0);
-
-      accumulateBindAnnotatedWith(
-          node, result, boundClass.getReceiver().getType().toString(), givenName.getValue());
+
+      CFValue value =
+          diATF
+              .getStoreBefore(node)
+              .getValue(JavaExpression.fromNode(methodInvocationNode.getTarget().getReceiver()));
+
+      if (value != null && !value.getAnnotations().isEmpty()) {
+        AnnotationMirror anno = value.getAnnotations().first();
+        List<String> classNames =
+            AnnotationUtils.getElementValueArray(anno, diATF.bindValValueElement, String.class);
+        MethodInvocationNode namesClassNode =
+            (MethodInvocationNode) methodInvocationNode.getArgument(0);
+        StringLiteralNode givenName = (StringLiteralNode) namesClassNode.getArgument(0);
+
+        accumulateBindAnnotatedWith(node, result, classNames.get(0), givenName.getValue());
+      }
+    }
+
+    return result;
+  }
+
+  @Override
+  public TransferResult<CFValue, CFStore> visitAssignment(
+      AssignmentNode node, TransferInput<CFValue, CFStore> input) {
```

Comment:
What's the purpose of this code? I don't think it should be necessary to handle dataflow through assignments, which should already be handled by `CFAbstractTransfer`. That is, all checkers automatically will transfer any information about the right-hand side of an assignment to the left-hand side: you don't need to write extra code to achieve that.

If this is doing something else, it needs a comment to explain its purpose.

Reply:
I had also assumed that the checker was going to automatically transfer any information about the right-hand side of an assignment to the left-hand side, but I wasn't seeing that behavior.
```java
      AnnotatedBindingBuilder<String> stringBinding = bind(String.class);
      stringBinding.toInstance("jdbc:mysql://localhost/pizza");
```
Before implementing `visitAssignment`, `stringBinding` _was not_ annotated with `@Bind()`